Penteract/Assets/Scripts/AIMovement.cpp

```diff
@@ -84,6 +87,7 @@ void AIMovement::Update() {
     case AIState::RUN:
         Seek(player->GetComponent<ComponentTransform>()->GetGlobalPosition(), maxSpeed);
         if (CharacterInMeleeRange(player)) {
+            agent->RemoveAgentFromCrowd();
```

Comment:
Is this agent validated to be assigned? There is a validation in the Start() but none in the update it might get to this point and still be nullptr

Reply:
Yes, I have realized this. I have already controlled all those cases in the branch where I am abstracting the AI movement behaviors and changing this script into MeleeGrunt.script.